btc/utils.go

```diff
@@ -1,29 +1,114 @@
 package btc
 
-import(
+import (
+	"bytes"
+	"encoding/hex"
 	"errors"
 
-	"github.com/btcsuite/btcutil/hdkeychain"
+	"github.com/tyler-smith/go-bip39"
+
+	"github.com/btcsuite/btcutil"
+
 	"github.com/btcsuite/btcd/chaincfg"
+	"github.com/btcsuite/btcd/txscript"
+	"github.com/btcsuite/btcutil/hdkeychain"
 )
 
+func seedFromMnemonic(mnemonic string) []byte {
+	return bip39.NewSeed(mnemonic, "")
+}
+
 func createWallet(network string) (*wallet, error) {
-	// generate a random seed
-	seed, err := hdkeychain.GenerateSeed(hdkeychain.RecommendedSeedLen)
-	if err!= nil {
+	// generate entropy for mnemonic
+	entropy, err := bip39.NewEntropy(EntropyBitSize)
+	if err != nil {
+		return nil, err
+	}
+	// generate mnemonic
+	mnemonic, err := bip39.NewMnemonic(entropy)
+	if err != nil {
 		return nil, err
 	}
 
-	hkStart := uint32(0x80000000)
+	// derivation path hard coded to m/44'/0'/0'/0 for mainnet
+	// and m/44'/1'/0'/0 for testnet as specified in BIP44
+	// support for BIP49 will be added later
+	// hardened key derivation starts at index 2147483648 (hex 0x80000000)
+	var path []uint32
+	if network == MainNet {
+		path = []uint32{Purpose, CoinTypeMainNet, Account, Change}
+	}
+	if network == TestNet {
+		path = []uint32{Purpose, CoinTypeTestNet, Account, Change}
+	}
```

Comment:
What happens if regtest? It will be recognized as testnet?

Reply:
no support for regtest at the moment, we should open an issue for this